subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -66,23 +65,28 @@ private void downloadInternal(URI address, File destination)
             in = conn.getInputStream();
             byte[] buffer = new byte[BUFFER_SIZE];
             int numRead;
-            int contentLength = conn.getContentLength();
-            long progressCounter = 0;
-            long numDownloaded = 0;
+            long totalLength = conn.getContentLengthLong();
```

Comment:
I believe this call is available from Java 7. [Docs](https://gradle.org/install/) say it is required, but I’m not sure if it is OK anyway.

> Gradle runs on all major operating systems and requires only a Java JDK or JRE version 7 or higher to be installed.

Reply:
nit: I'd keep `contentLength` name tho, it's a common terminology

Reply:
`total` and `downloaded` are content length both, I’ve just trimmed it down from `totalContentLength` and `downloadedContentLength` since it is pretty obvious what this function does.

Reply:
So, detect JVM version and use `getContentLengthLong()` if it's Java 7+ otherwise `getContentLength()`?

Reply:
Is there any reason we cannot continue using `getContentLength()`? I think 2GB is a reasonable upper bound for a Gradle distribution. After that, well I imagine we'll have moved past Java 6 in all forms :)